src/object/obj_ec.h

```diff
@@ -753,8 +754,16 @@ obj_ec_parity_lists_match(struct daos_recx_ep_list *lists_1,
 			     list_2->re_items[j].re_recx.rx_idx) ||
 			    (list_1->re_items[j].re_recx.rx_nr !=
 			     list_2->re_items[j].re_recx.rx_nr)) {
-				D_ERROR("got different parity recx in EC data recovery\n");
-				return -DER_IO;
+				if (!DAOS_RECX_OVERLAP(list_1->re_items[j].re_recx,
+						       list_2->re_items[j].re_recx)) {
+					D_ERROR("got different parity recx in EC data recovery\n");
```

Comment:
Hmm, how about this case?
list1 [0-2M], [4-6M]
list2 [1M-3M], [5M-7M] 

should return -DER_IO ?

Reply:
I think that case is failure case, because it happens when client try to recover one or multiple full-stripe, so only reasonable mismatch is that one parity return multiple partial exts, and another return less number of merged exts, right?

Reply:
I mean in this case, if parity1 missing [2M-4M], [6M-7M], parity2 missing [0-1M], [3M-5M] due to the corruption, then it should return EIO, instead of keep retrying?

Reply:
that case current code will return EIO, no?